src/dns/coredns.rs

```diff
@@ -140,23 +140,27 @@ impl CoreDns {
                             let mut resolved_ip_list: Vec<IpAddr> = Vec::new();
                             let mut nameservers_scoped: Vec<ScopedIp> = Vec::new();
                             // Add resolvers configured for container
-                            if let Some(Some(dns_servers)) = backend.ctr_dns_server.get(&src_address.ip()) {
-                                    if !dns_servers.is_empty() {
-                                        for dns_server in dns_servers.iter() {
-                                            nameservers_scoped.push(ScopedIp::from(*dns_server));
-                                        }
+                            if backend.ctr_is_internal(&src_address.ip()) {
```

Comment:
would it make more sense to move this below to the  
"Not found, forwarding dns request for" case?

Also what is the expected behavior for a internal container, should it get NXDomain? What happens if no upstream resolvers are tried here? Do we just not answer at all and let the client timeout?

Reply:
Internal containers should still respond correctly (assuming same network).
Unfortunately it looks like we're timing out instead of NXDOMAIN-ing. I think we can probably detect the 0 resolvers in use case and manually NXDOMAIN instead of sending it out to 0 resolvers and timing out... Let me fix that

Reply:
there is a no_proxy case already so you can just reuse it for internal as well